pkg/apply/taskrunner/context.go

```diff
@@ -242,3 +346,11 @@ type applyInfo struct {
 	// uid captures the uid of the resource that has been applied.
 	uid types.UID
 }
+
+// deleteInfo captures information about resources that have been
+// deleted. This is captured in the TaskContext so other tasks
+// running later might use this information.
+type deleteInfo struct {
+	// uid captures the uid of the resource that has been applied.
+	uid types.UID
```

Comment:
Unfortunately `Generation` is not available from the DELETE response. If the object delete was delayed (deleteTimestamp) then the http call returns a non-KRM status object, not the object itself. And the dynamic client doesn't return anything but an error. So if you want the object when it does return you need to use the RESTClient directly...

This needs to be changed in upstream K8s, but it's not gonna happen any time soon.